labs/a10.md

```diff
@@ -0,0 +1,369 @@
+Unfortunately, both the CEO and CTO are busy handling the business side, which leaves it up to you to get their web 
+application deployed using Docker and `docker-compose`.
+
+Don't worry though -- while you may not have health insurance or a nice salary, you do have some of the CTO's notes and 
+equity to help you with your task. You get off BART at 12pm and enter your cramped SOMA coworking space, sitting down 
+at the desk you share with the CTO while cracking open a cold LaCroix. Checking your email, you find the following 
+notes from the CTO:
+
+#### About `docker-compose`
+`docker-compose` lets you define applications that require more than one container to function. For example, on a web 
+application you may want your actual web application running inside of a single container, and your database running in 
+a different container.
+
+Typically you define applications in terms of **services**. Again, going with the web application example, there are 
+two distinct services: the app itself, and the database backing it. `docker-compose` lets you define different services 
+within a [YAML](https://en.wikipedia.org/wiki/YAML) file and run the services accordingly. 
+
+One of the nice things about `docker-compose` is that it automatically sets up a network for your containers in which:
+- each container for a service is on the network and reachable from other containers on the network
+- each container is discoverable on the network via its container name
+
+This means it should be pretty simple to get our web app to connect to the database.
+
+#### Installing
+`docker-compose` is really just a python package, so ideally we should just install it with `virtualenv` and `pip`. 
+Feel free to do it this way if you want, but since we're a hotshot tech startup we're gonna do this the cowboy way 
+(yeehaw):
+
+```
+sudo curl -L https://github.com/docker/compose/releases/download/1.20.1/docker-compose-`uname -s`-`uname -m` -o 
+/usr/local/bin/docker-compose
+sudo chmod +x /usr/local/bin/docker-compose
+```
+
+#### The web application
+The web application is written only the most badass rockstar tech, 
+[Node.js](https://www.youtube.com/watch?v=bzkRVzciAZg). For the database, it uses the most webscale, reliable, and 
+persistent database available on the market today, [MongoDB](https://www.youtube.com/watch?v=b2F-DItXtZs). _Amazing_, 
+you think to yourself, _every word of what he just said was wrong_.
+
+The web application can be found on GitHub. Note that the web app listens on port 8080, so you'll need to forward or 
+expose that port. Instructions for setting it up are located in the `README.md` of the repository:
+
+`https://github.com/0xcf/decal-sp18-a10`
+
+For MongoDB, you can just use the image on [DockerHub](https://hub.docker.com/_/mongo/) (a website where people can 
+upload built Docker images). It's just called `mongo`. For example, if you wanted to run MongoDB within a container in 
+detached mode:
+
+`docker run -d mongo:latest`
```

Comment:
make some note that in the real world we probably woudl just specify version # instead of using latest